test/datalog_tests/simple2.dl

```diff
@@ -471,3 +471,129 @@ FlatGroup(x) :-
     GroupThis(x, y),
     var g = x.group_by(y),
     var x = FlatMap(g).
+
+/*
+ * for-loop patterns
+ */
+
+typedef ForStruct1<'T> = ForStruct1 {
+    f1: 'T,
+    f2: istring
+}
+
+typedef ForStruct2 = ForStruct2 {
+    f1: (bool, double),
+    f2: ForStruct1<(usize, usize)>
+}
+
+function testMap(): Map<(usize, string), ForStruct2> {
```

Comment:
what is the difficulty in creating a `const` construct in DDlog that is mapped to a global constant in Rust?

Reply:
Shouldn't be too hard.  One caveat is that the parse won't be able to disambiguate between constructor names and constants in some cases, so we'll have to resolve this at a later stage, but we already have similar logic for variable name vs function name resolution, so this is not a problem.